### PR TITLE
[JBIDE-25670] remove all resources in a fixed order

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@jkopriva can you please review this since you reported the issue? thx!

**Comment by adietish** [comment]
jenkins is broken with some connection reset error :(

**Comment by jkopriva** [comment]
testPR

**Comment by adietish** [comment]
@jkopriva can you review this please?

File: tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/internal/core/job/DeleteResourcesJobTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
```
Comment (jkopriva) [inline tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/internal/core/job/DeleteResourcesJobTest.java]: @adietish Copyright year should be changed to 2018

File: tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/internal/core/job/DeleteResourcesJobTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IPod;
+import com.openshift.restclient.model.IReplicationController;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.secret.ISecret;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (jkopriva) [inline tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/internal/core/job/DeleteResourcesJobTest.java]: @adietish I would also add the author, even if it is only a test class.

**Comment by jkopriva** [comment]
@adietish I have added comments, otherwise it works as expected and we can merge this.

**Comment by adietish** [comment]
@jkopriva great, thx!

**Comment by adietish** [comment]
testPR

**Pull request merged**